src/dsc.h

```diff
@@ -467,16 +483,13 @@ class IdList {
         // Look to see if we already have something with the same handle value.
         ssassert(FindByIdNoOops(t->h) == nullptr, "Handle isn't unique");
 
-        // Find out where the added element should be.
+        // Shift indexes to the end of the array, if needed.
         int pos = LowerBoundIndex(*t);
+        memmove(indexes + pos + 1, indexes + pos, sizeof *indexes * (n - pos));
```

Comment:
I wonder if [insert in a `vector<int>`](https://github.com/ruevs/solvespace/blob/6bc1aa91c6a0f1acfc986b34ff54ed36efcdbfc1/src/dsc.h#L537) is as fast. I'll debug it tomorrow to see what it does.

Reply:
It should optimize that way.